tests/geometry/test-distance.cpp

```diff
@@ -1087,6 +1089,107 @@ void test_segment_triangle_distance()
 }
 
+void test_point_ellipse_distance_2d()
+{
+    const geode::Point2D center{ { 0.0, 0.0 } };
+    const geode::Vector2D first_axis{ { 3.0, 0.0 } };
+    const geode::Vector2D second_axis{ { 0.0, 2.0 } };
+    const geode::Frame2D frame{ { first_axis, second_axis } };
+    const geode::Ellipse2D ellipse{ center, frame };
+    const geode::Point2D on_boundary{ { 3.0, 0.0 } };
+    const geode::Point2D outside1{ { 4.0, 0 } };
+    const geode::Point2D outside2{ { 0, 5.0 } };
+    const auto [on_boundary_distance, on_boundary_closest_point] =
+        geode::point_ellipse_distance( on_boundary, ellipse );
+    OPENGEODE_EXCEPTION(
+        on_boundary_distance == 0. && on_boundary_closest_point == on_boundary,
+        "[Test] Wrong result for point_ellipse_distance_2d with on_boundary "
+        "point" );
+    const auto [outside1_distance, outside1_closest_point] =
+        geode::point_ellipse_distance( outside1, ellipse );
+    const geode::Point2D result_outside1{ { 3, 0 } };
+    OPENGEODE_EXCEPTION(
+        outside1_distance == 1
+            && outside1_closest_point.inexact_equal( result_outside1 ),
+        "[Test] Wrong result for point_ellipse_distance_2d with outside1 "
+        "point" );
+    const auto [outside2_distance, outside2_closest_point] =
+        geode::point_ellipse_distance( outside2, ellipse );
+    const geode::Point2D result_outside2{ { 0, 2 } };
+    OPENGEODE_EXCEPTION(
+        outside2_distance == 3
+            && outside2_closest_point.inexact_equal( result_outside2 ),
+        "[Test] Wrong result for point_ellipse_distance_2d with outside2 "
+        "point" );
+}
+
+void test_point_ellipse_distance_2d_not_aligned()
+{
+    const geode::Point2D center{ { 0.0, 0.0 } };
+    const geode::Vector2D first_axis{ { 1.0, 1.0 } };
+    const geode::Vector2D second_axis{ { -1.0, 1.0 } };
+    const geode::Frame2D frame{ { first_axis, second_axis } };
+    const geode::Ellipse2D ellipse{ center, frame };
+    const geode::Point2D point{ { 2.0, 2.0 } };
+    const auto [distance, closest_point] =
+        geode::point_ellipse_distance( point, ellipse );
+    const geode::Point2D result{ { 1.0, 1.0 } };
+    OPENGEODE_EXCEPTION( closest_point.inexact_equal( result ),
+        "[Test] Wrong result for point_ellipse_distance_2d_not_aligned" );
+}
+
+void test_point_ellipse_distance_3d()
+{
+    const geode::Point3D center{ { 0.0, 0.0, 0.0 } };
+    const geode::Vector3D first_axis{ { 3.0, 0.0, 0.0 } };
+    const geode::Vector3D second_axis{ { 0.0, 2.0, 0.0 } };
+    const geode::Vector3D third_axis{ { 0.0, 0.0, 1.0 } };
+    const geode::Frame3D frame{ { first_axis, second_axis, third_axis } };
+    const geode::Ellipse3D ellipse{ center, frame };
+    const geode::Point3D on_boundary{ { 3.0, 0.0, 0.0 } };
+    const geode::Point3D outside1{ { 4.0, 0, 0 } };
+    const geode::Point3D outside2{ { 0, 5.0, 0 } };
+    const geode::Point3D outside3{ { 0, 0, 3 } };
```

Comment:
Maybe you also can try one point inside to see if there is a probleme or not in distance computation